### PR TITLE
New: Add update functions for SDS

### PR DESCRIPTION
**Pull request by rounak-adhikary**
# Description
New functionality for SDS:
1. The create function is enhanced to support all params.
2. The sds read has been fixed. Earlier, the list of ip addresses was not getting unmarshalled from REST API json into sds struct.
3. Added functions to update ip addresses, rmcache, rfcache, port, drlmode and performance profile.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Most of the functionality has been tested by:
1. Test function in separate branch https://github.com/dell/goscaleio/blob/sds-tf-test/inttests/sds_test.go#L149
```
root@lglap049:~/goscaleio# make int-test
The SDC is not installed
All of the drv_cfg tests cannot be run
Starting tests
=== RUN   TestCreateSdsParamsInvalid
--- PASS: TestCreateSdsParamsInvalid (0.42s)
=== RUN   TestCreateSdsParams
    sds_test.go:178: The number of I/O buffers is 0
    sds_test.go:179: The port is 7072
    sds_test.go:180: The rmcacheenabled is false
    sds_test.go:181: The rmcachesize in kb is 155648
    sds_test.go:182: The ip list is {IP: 10.247.100.232, role: all},{IP: 0.2.2.2, role: sdcOnly},
    sds_test.go:183: The drl mode is Volatile
    sds_test.go:184: The perf profile is HighPerformance
    sds_test.go:185: The Rf Cache enablement is false
    sds_test.go:186: The Fgl Metadata Cache size is 0
    sds_test.go:213: Updation done ============
    sds_test.go:214: The number of I/O buffers is 0
    sds_test.go:215: The port is 7072
    sds_test.go:216: The rmcacheenabled is true
    sds_test.go:217: The rmcachesize in kb is 262144
    sds_test.go:218: The ip list is {IP: 10.247.100.232, role: all},{IP: 0.2.2.2, role: sdcOnly},{IP: 0.2.2.3, role: sdcOnly},
    sds_test.go:219: The drl mode is NonVolatile
    sds_test.go:220: The perf profile is Compact
    sds_test.go:221: The Rf Cache enablement is true
    sds_test.go:222: The Fgl Metadata Cache size is 0
--- PASS: TestCreateSdsParams (1.66s)
PASS
coverage: 21.2% of statements in github.com/dell/goscaleio
ok      github.com/dell/goscaleio/inttests      2.122s  coverage: 21.2% of statements in github.com/dell/goscaleio
root@lglap049:~/goscaleio# 
```
2. By using this in terraform CRUD functions https://github.com/dell/terraform-provider-powerflex/tree/sds-res-fix

### CODE REVIEW

File: inttests/sds_test.go [hunk 1]
```diff
@@ -124,23 +125,35 @@ func TestCreateSdsInvalid(t *testing.T) {
...
 
 // TestCreateSdsInvalid will attempt to add an SDS, which results in failure
-func TestCreateSdsIPRoleInvalid(t *testing.T) {
+func TestCreateSdsParamsInvalid(t *testing.T) {
 	pd := getProtectionDomain(t)
 	assert.NotNil(t, pd)
 
 	// attempt to create an SDS with a number of invalid IPs
 	// this is done, in a failure mode, to prevent changing the Protection Domain used for testing
 	sdsName := "invalid"
```
Comment (khareRajshree) [inline inttests/sds_test.go:134]: Hi @rounak-adhikary, can we have some mock values that tests against valid test cases as well.
Also applicable to other tests methods where only invalid scenarios are tested.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

File: inttests/sds_test.go (hunk 1 above)
Comment (rounak-adhikary) [reply inttests/sds_test.go:134]: @khareRajshree For valid testcases for SDS we need atleast one real IP. Is there a dedicated IP that I can use? I can reuse this testcase (https://github.com/dell/goscaleio/blob/sds-tf-test/inttests/sds_test.go#L149) that I was using.

Comment (rounak-adhikary) [reply inttests/sds_test.go:134]: I believe that the lack of dedicated IPs is what led to the existence of only invalid testcases for SDS.

Comment (khareRajshree) [reply inttests/sds_test.go:134]: yea that's also true, we can leave the valid TC for now, as we cannot put the internal systems IPs out here.

**Review by khareRajshree — APPROVED** [review]
LGTM

**Review by ashleyvjoy — APPROVED** [review]

**Pull request merged**